deny.toml

```diff
@@ -44,6 +44,7 @@ allow = [
     "MPL-2.0",
     "Unlicense",
     "Unicode-3.0",
+    "Zlib",
```

Comment:
do we need to adopt this license? If so I'd suggest we do it in a different PR as we did with the Unicode one on  https://github.com/libp2p/rust-libp2p/pull/5738

Reply:
Reverted. Please note that CI fails without this change: https://github.com/libp2p/rust-libp2p/actions/runs/12519077957/job/34922566527?pr=5755#step:4:25868

Reply:
no worries, we can address it on a subsequent PR :)